### ISSUE TITLE
Title is not announced by Voiceover when Dialog modal initially opened

### ISSUE THREAD

**Issue by mchensd** [radix-ui/primitives#3348]
### Current Behavior

When opening the Radix Dialog, the title is not announced by Mac's Voiceover. Example is on the Radix  [docs](https://www.radix-ui.com/primitives/docs/components/dialog). We are seeing similar behavior in our component built on top of Radix Dialog.

<img width="758" alt="Image" src="https://github.com/user-attachments/assets/7d863380-a156-440f-841b-7c6269bf159f" />

### Expected behavior

Expected the title 'Edit profile' to be announced in the example above.

### Reproducible example

Happens when clicking 'Edit profile' on the example at https://www.radix-ui.com/primitives/docs/components/dialog

Mac OS version: Sonoma 14.7.2

### Your environment

| Software         | Name(s) | Version |
| ---------------- | ------- | ------- |
| Radix Package(s) |  Dialog       |         |
| React            | n/a     |         |
| Browser          |    Chrome     |     132.0.6834.84    |
| Assistive tech   |    Mac Voiceover     |    Sonoma 14.7.2     |
| Node             | n/a     |         |
| npm/yarn         |         |         |
| Operating System |  Mac       |     Sonoma 14.7.2    |

**Comment by grelas** [comment]
Just to add, with Mac Voiceover on Monterey 12.7.1, it announces the modal title:
<img width="765" alt="Image" src="https://github.com/user-attachments/assets/39d42f4c-0749-4b73-bf29-e5197d64f96a" />

**Comment by grelas** [comment]
Seems to work on Safari, not Chrome, so maybe a chrome-specific bug?

**Comment by shamokit** [comment]
I've run into the same issue: the dialog title (referenced via aria-labelledby) isn't being announced, but only in Chrome on macOS.
It works as expected in Safari and Firefox, so this seems like a Chrome-specific bug.
I’ve kind of given up on it for now, but would love to hear if anyone has a workaround.

**Comment by tpommellet-pass** [comment]
same issue here : anyone got a solution ?
